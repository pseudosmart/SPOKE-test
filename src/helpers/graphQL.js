import { PRODUCTS } from '../constants/graphQLFragments'

export function query() {
  return fetch('https://spoke-3.myshopify.com/api/2020-04/graphql.json', {
    headers: {
      'Accept': 'application/json',
      'Content-Type': 'application/json',
      'X-Shopify-Storefront-Access-Token': '2dc239eb7f8edb6219b5abc7bc0bc361'
    },
    credentials: 'same-origin',
    method: 'POST',
    body: JSON.stringify({ query: PRODUCTS })
  }).then(res => res.json())
}

export default {
  query
}
