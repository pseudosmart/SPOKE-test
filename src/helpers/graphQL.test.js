require('jest-fetch-mock').enableMocks()
import graphQL from './graphQL'

test('it can send a request to the server', () => {
  fetch.mockResponseOnce(JSON.stringify({ data: {}}))

  return graphQL.query('{ getTraveller { id } }')
    .then(() => {
      expect(fetch).toMatchSnapshot()
    })
})
