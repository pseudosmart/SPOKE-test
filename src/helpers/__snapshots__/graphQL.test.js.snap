// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it can send a request to the server 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "https://spoke-3.myshopify.com/api/2020-04/graphql.json",
      Object {
        "body": "{\\"query\\":\\"{\\\\n  products(first: 50, query:\\\\\\"product_type:Heroes OR product_type:Sharps OR product_type:Polo\\\\\\") {\\\\n    edges {\\\\n      node {\\\\n        id,\\\\n        \\\\n  images(first: 1) {\\\\n    edges {\\\\n      node {\\\\n        originalSrc\\\\n      }\\\\n    }\\\\n  }\\\\n\\\\n        productType,\\\\n        title,\\\\n        tags,\\\\n        \\\\n  priceRange {\\\\n    maxVariantPrice {\\\\n      amount\\\\n    }\\\\n  }\\\\n\\\\n        updatedAt,\\\\n      }\\\\n    }\\\\n  }\\\\n}\\\\n\\"}",
        "credentials": "same-origin",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
          "X-Shopify-Storefront-Access-Token": "2dc239eb7f8edb6219b5abc7bc0bc361",
        },
        "method": "POST",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
